### PR TITLE
Fix Travis CI issues

### LINKED ISSUE #1: Fix Travis CI
>error eslint@6.3.0: The engine "node" is incompatible with this module. Expected version "^8.10.0 || ^10.13.0 || >=11.10.1". Got "9.11.2"

### PR DESCRIPTION
**Pull request by BronsonQuick**
Fixes #1.

### CODE REVIEW

**Comment by BronsonQuick** [comment]
@rmccue I could use an assist on this one please as I need to brush up on my TypeScript. If you run `yarn build` you'll see the errors trickle in.

File: src/KeyHandler.tsx [hunk 1]
```diff
@@ -24,11 +24,11 @@ export default class KeyHandler extends React.Component<Props> {
 	static propTypes = {
 		disableHint: PropTypes.bool,
 		enabled: PropTypes.bool,
-		shortcut: PropTypes.string.isRequired,
+		shortcut: PropTypes.string,
 		onTrigger: PropTypes.func,
 	};
 
-	handler: number;
+	handler: number | null;
```
Comment (rmccue) [inline src/KeyHandler.tsx]: This should be `handler?: number;`

File: src/context.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
 
-const KeyContext = React.createContext( undefined );
+const KeyContext = React.createContext( {} );
```
Comment (rmccue) [inline src/context.ts]: This is incorrect, I specifically want `undefined` as the value.

File: src/keys.ts [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
 	id: number;
 	callback: Callback;
-};
+}
 
 const toCode = ( text: string ) : KeyCode => {
 	const keys = text.split( '+' ).map( key => key.trim() );
 	const code = {
-		key: null,
+		key: string,
```
Comment (rmccue) [inline src/keys.ts]: This is incorrect, the value should be `null`.

File: src/keys.ts [hunk 4]
```diff
@@ -69,7 +71,7 @@ const toCode = ( text: string ) : KeyCode => {
 				if ( code.key === key.toUpperCase() ) {
 					// Ignore shift status for symbols, and just map to the
 					// value. This should work reliably on all keyboards.
-					code.shiftKey = null;
+					code.shiftKey = false;
```
Comment (rmccue) [inline src/keys.ts]: This is incorrect, the value should be `null` for the check below in `compare`.

File: tsconfig.json [hunk 5]
```diff
@@ -3,22 +3,34 @@
...
-    // "noUnusedLocals": true,
-    // "noUnusedParameters": true
+    "allowSyntheticDefaultImports": true,
+    "skipLibCheck": true,
+    "esModuleInterop": true,
+    "strict": true,
+    "forceConsistentCasingInFileNames": true,
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "noEmit": true
```
Comment (rmccue) [inline tsconfig.json]: Most of these are incorrect; the automated changes by `react-scripts test` make a lot of assumptions. We need to turn most of these back off.

Comment (BronsonQuick) [reply tsconfig.json]: Ahh right. I didn't think you'd make so many errors which is kinda why I started then gave up to get some feedback. Taa!

I don't supposed you have an example tsconfig.json for your or HM's coding standards that we could use? I'm keen to level up my TypeScript a little!

Comment (rmccue) [reply tsconfig.json]: Yeah, the example tsconfig.json is the one that's committed 😉

**Pull request closed without merging**